lxd/devlxd.go

```diff
@@ -118,17 +118,19 @@ func devlxdConfigKeyGetHandler(d *Daemon, c instance.Instance, w http.ResponseWr
 		return response.DevLxdErrorResponse(api.StatusErrorf(http.StatusForbidden, "not authorized"), c.Type() == instancetype.VM)
 	}
 
-	value := c.ExpandedConfig()[key]
+	var value string
 
-	// If parsing the config is not possible, abstain from merging the additional keys into the config.
-	if strings.HasSuffix(key, ".vendor-data") || strings.HasSuffix(key, ".user-data") {
-		value, err = util.MergeSSHKeyCloudConfig(c.ExpandedConfig(), value)
-		if err != nil {
-			logger.Warn("Failed merging SSH keys into cloud-init seed data, abstain from injecting additional keys", logger.Ctx{"err": err, "project": c.Project().Name, "instance": c.Name(), "requestedKey": key})
-		}
+	// For values containing cloud-init seed data, try to merge into them additional SSH keys present on the instance config.
+	// If parsing the config is not possible, abstain from merging the additional keys.
+	if strings.HasSuffix(key, ".user-data") {
```

Comment:
Please can we do exact matching on the relevant keys rather than suffix matching otherwise customer keys like user.foo.user-data would be in scope incorrectly.